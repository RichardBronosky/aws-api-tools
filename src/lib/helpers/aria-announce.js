const formatMsg = msg => {
  return '<div aria-live="assertive" aria-atomic="true">' + msg + '</div>'
}

export const taskDone = (fn, time, id) => {
  let timers = {}
  if (typeof fn == 'function') {
    if (timers[id]) clearTimeout(timers[id])
    timers[id] = setTimeout(fn, time)
  }
}

export const announce = (msg, form, ele, delay) => {
  if (typeof ele == 'object') {
    msg += ele.innerText.replace(/\n/g, ' ')
  }
  if (document) {
    document.getElementById('aria-announce').innerHTML = ''
    taskDone(
      () => {
        document.getElementById('aria-announce').innerHTML = formatMsg(msg)
      },
      delay || 500,
      'announceMessage'
    )
    if (form) {
      taskDone(
        () => {
          form.querySelectorAll('[aria-invalid="true"]')[0].focus()
        },
        1000,
        'focusElement'
      )
    }
  }
}
