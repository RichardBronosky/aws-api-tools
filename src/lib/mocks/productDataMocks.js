export const mockProduct1 = {
  addon_items: [
    {
      category: 'bedroom',
      delivery_type: 'K',
      list_price: 49.99,
      price: 49.99,
      quantity: 2,
      sku: '50222606',
      title: 'Twin Bunkie Boards',
      addon_required: true,
    },
  ],
  free_shipping: true,
  variation_keys: [],
  type: 'item',
  title: 'Adaire Brown Side Table',
  description:
    "Legend has it that King Arthur's sword Excalibur was forged on a mythical islanda place where grapevines were said to grow amongst wild apple trees. This meeting of wood and metal was the inspiration behind our bold new Adaire side table. Each piece is crafted with slats of acacia wood in angled patterns, creating a rustic look offset by modern metal legs and sleek handles. Customer assembly is required. Customer assembly is required.",
  delivery_type: 'O',
  catalog: 'adult',
  room_type_code: 'LR',
  single_item_room: false,
  sell_individually: true,
  mpn: '231-CS860',
  swatch_request_flag: false,
  generic_name: 'side table',
  monogram: false,
  decor: ['contemporary'],
  finish_family: ['dark wood'],
  dimensions: '23"l x 23"w x 24"h',
  finish: ['brown'],
  special_offer: true,
  style: ['square'],
  catalog_availability: { TX: true, FL: true, OOM: true, SE: true },
  category: 'tables',
  sub_category: ['End Tables'],
  pricing: {
    SE_3_list_price: 349.99,
    SE_3_sale_price: 279.99,
    default_price: 349.99,
    FL_0_list_price: 349.99,
    FL_0_sale_price: 279.99,
    FL_6_list_price: 349.99,
    FL_6_sale_price: 279.99,
    TX_5_list_price: 349.99,
    TX_5_sale_price: 279.99,
    SE_0_list_price: 349.99,
    SE_0_sale_price: 279.99,
    SE_4_list_price: 349.99,
    SE_4_sale_price: 279.99,
    TX_4_list_price: 349.99,
    TX_4_sale_price: 279.99,
    TX_6_list_price: 349.99,
    TX_6_sale_price: 279.99,
    FL_4_list_price: 349.99,
    FL_4_sale_price: 279.99,
    FL_3_list_price: 349.99,
    FL_3_sale_price: 279.99,
    TX_0_list_price: 349.99,
    TX_0_sale_price: 279.99,
    SE_2_list_price: 349.99,
    SE_2_sale_price: 279.99,
  },
  on_sale: {
    SE_3: true,
    FL_0: true,
    FL_6: true,
    TX_5: true,
    SE_0: true,
    SE_4: true,
    TX_4: true,
    TX_6: true,
    FL_4: true,
    FL_3: true,
    TX_0: true,
    SE_2: true,
  },
  strikethrough: { FL: true, OM: true, SE: true, TX: true },
  sku_promotion_description: {
    FL: 'Memorial Day Drop Ship',
    OM: 'Memorial Day Drop Ship',
    SE: 'Memorial Day Drop Ship',
    TX: 'Memorial Day Drop Ship',
  },
  ad_position: { FL: '0', OM: '0', SE: '0', TX: '0' },
  shipping_cost_code: { FL: 'VD71', SE: 'VD71', TX: 'VD71' },
  breadcrumb_url: '/furniture/living-rooms',
  you_may_also_like_items: [
    {
      sku: '22122369',
      title: 'Adaire Brown Cocktail Table',
      quantity: 1,
      label: 'you may also like',
      addon_required: false,
      pricing: {
        SE_4_list_price: 549.99,
        SE_4_sale_price: 549.99,
        default_price: 549.99,
        FL_4_list_price: 549.99,
        FL_4_sale_price: 549.99,
        SE_3_list_price: 549.99,
        SE_3_sale_price: 549.99,
        TX_6_list_price: 549.99,
        TX_6_sale_price: 549.99,
        SE_0_list_price: 549.99,
        SE_0_sale_price: 549.99,
        TX_4_list_price: 549.99,
        TX_4_sale_price: 549.99,
        SE_2_list_price: 549.99,
        SE_2_sale_price: 549.99,
        FL_3_list_price: 549.99,
        FL_3_sale_price: 549.99,
        FL_6_list_price: 549.99,
        FL_6_sale_price: 549.99,
        FL_0_list_price: 549.99,
        FL_0_sale_price: 549.99,
        TX_0_list_price: 549.99,
        TX_0_sale_price: 549.99,
        TX_5_list_price: 549.99,
        TX_5_sale_price: 549.99,
      },
      catalog_availability: { SE: true, TX: true, FL: true, OOM: true },
      delivery_type: 'O',
      category: 'tables',
      sell_individually: true,
    },
  ],
  primary_image:
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090',
  grid_image:
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_grid-image-item?cache-id=5414146faa5973392eccec06f9ab3e31',
  high_res_image:
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_high-res-image-item?cache-id=9d4049916c821c28a12bfd9438164d4e',
  swatch_image:
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_swatch-image-item?cache-id=462c75c04a46ff84dc67a370bddee768',
  alternate_images: [
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_alt-image-1?cache-id=d9eb8ed838154a8d6d59e699d3a8f94a',
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_alt-image-2?cache-id=fade3f3f0020c0462b73618a3de9ced8',
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_alt-image-3?cache-id=0dc792fe59668f81446dde858d303efa',
  ],
  alternate_grid_images: [
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_alt-grid-image-1?cache-id=72f93b04e81c83e6772bac543445fba2',
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_alt-grid-image-2?cache-id=f91b555873b9780b60143b3d7a3cc826',
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_alt-grid-image-3?cache-id=068364c441833b95ca6189391cd5dec5',
  ],
  lastModified: '2019-05-23T16:13:30.917Z',
  createdAt: '2019-05-23T16:13:30.917Z',
  sku: '23122360',
}

export const mockProduct2 = {
  free_shipping: false,
  variation_keys: ['A2C6BF7083B796559132D28DC0A8228994B4A99F'],
  type: 'room',
  title: 'Laney Park Dark Gray 7 Pc Sectional',
  description:
    'The Laney Park sectional is packed with features meant to enhance your lounging experience. Each oversized seat has a smooth adjusting back mechanism for customizable seat depth, making it the perfect place for entertaining friends or relaxing with family. Contemporary and stylish, the dark gray color and flat woven textured fabric are enhanced with clean, tailored lines. 100% Polyester.',
  delivery_type: 'D',
  catalog: 'adult',
  room_type_code: 'LR',
  single_item_room: true,
  sell_individually: true,
  mpn: '114-RG270',
  swatch_request_flag: false,
  generic_name: '7 pc sectional',
  monogram: false,
  color_family: ['gray'],
  material: ['textured'],
  movement: ['stationary'],
  style: ['sectional'],
  decor: ['contemporary'],
  dimensions: '116"w x 135"d x 33"h',
  color: ['dark gray'],
  material_family: 'textured',
  piececount: 7,
  catalog_availability: { FL: true, OOM: false, SE: true, TX: true },
  category: 'livingroom',
  sub_category: ['Living Room Sets', 'Sectionals'],
  package_skus: {
    SE: [
      { sku: '12216760', quantity: '1', price: '208.43' },
      { sku: '12116768', quantity: '4', price: '172.94' },
      { sku: '12316762', quantity: '1', price: '231.63' },
      { sku: '12516766', quantity: '1', price: '68.17' },
    ],
    TX: [
      { sku: '12316762', quantity: '1', price: '231.63' },
      { sku: '12216760', quantity: '1', price: '208.43' },
      { sku: '12516766', quantity: '1', price: '68.17' },
      { sku: '12116768', quantity: '4', price: '172.94' },
    ],
    FL: [
      { sku: '12316762', quantity: '1', price: '248.65' },
      { sku: '12116768', quantity: '4', price: '185.62' },
      { sku: '12216760', quantity: '1', price: '223.71' },
      { sku: '12516766', quantity: '1', price: '73.16' },
    ],
  },
  pricing: {
    SE_0_list_price: 1299.99,
    SE_0_sale_price: 1199.99,
    default_price: 1388,
    TX_0_list_price: 1388,
    TX_0_sale_price: 1199.99,
    FL_6_list_price: 1199.99,
    FL_6_sale_price: 1199.99,
    TX_6_list_price: 1388,
    TX_6_sale_price: 1199.99,
    TX_5_list_price: 1388,
    TX_5_sale_price: 1199.99,
    SE_3_list_price: 1199.99,
    SE_3_sale_price: 1199.99,
    FL_4_list_price: 1199.99,
    FL_4_sale_price: 1199.99,
    FL_3_list_price: 1299.99,
    FL_3_sale_price: 1288,
    SE_2_list_price: 1299.99,
    SE_2_sale_price: 1199.99,
    TX_4_list_price: 1388,
    TX_4_sale_price: 1199.99,
    SE_4_list_price: 1199.99,
    SE_4_sale_price: 1199.99,
    FL_0_list_price: 1299.99,
    FL_0_sale_price: 1288,
  },
  on_sale: {
    SE_0: true,
    TX_0: true,
    FL_6: false,
    TX_6: true,
    TX_5: true,
    SE_3: false,
    FL_4: false,
    FL_3: true,
    SE_2: true,
    TX_4: true,
    SE_4: false,
    FL_0: true,
  },
  shipping_cost_code: { FL: 'ROOM', SE: 'ROOM', TX: 'ROOM' },
  is_featured: { FL: true, SE: true, TX: true },
  variations: {
    color: [
      {
        sku: '1141677P',
        title: 'Laney Park Dark Gray 7 Pc Sectional',
        catalog_availability: { FL: true, OOM: false, SE: true, TX: true },
        variation_value: 'Dark Gray',
        image:
          'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_swatch-image-room?cache-id=b574d3b4897f3e9848ffdec84989abe3',
      },
      {
        sku: '1141676P',
        title: 'Laney Park Light Gray 7 Pc Sectional',
        catalog_availability: { TX: true, FL: true, OOM: false, SE: true },
        variation_value: 'Light Gray',
        image:
          'http://images.rtg-dev.com/laney-park-light-gray-7-pc-sectional_1141676P_swatch-image-room?cache-id=4623cf4bc97371730ba3c414794da78c',
      },
    ],
  },
  breadcrumb_url: '/furniture/living-rooms',
  primary_image:
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc',
  grid_image:
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_grid-image-room?cache-id=ae141488ad62b690ff8edb6ba7635c0a',
  high_res_image:
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_high-res-image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc',
  swatch_image:
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_swatch-image-room?cache-id=b574d3b4897f3e9848ffdec84989abe3',
  alternate_images: [
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-image-1?cache-id=bec562965fec72af770c2db65c72e1ba',
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-image-2?cache-id=6035fedf0cf68d55a8b118999fbb2bb5',
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-image-3?cache-id=322a89d2f5eb4b25208d1beebc4c5d25',
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-image-4?cache-id=91df28c6afd4eb7720750245bfa2d131',
  ],
  alternate_grid_images: [
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-grid-image-1?cache-id=a638c2b14ccff7a4e9af1100ea21ed52',
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-grid-image-2?cache-id=24dc7ae0b65f2fc9c3d7a18a122aca22',
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-grid-image-3?cache-id=cb24d99a79ab1614820e9b8b044bdc48',
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-grid-image-4?cache-id=898ccb8827b2499b3893f69959215f22',
  ],
  lastModified: '2019-05-23T16:16:03.416Z',
  createdAt: '2019-05-23T16:16:03.416Z',
  sku: '1141677P',
}

export const mockAddonItem = {
  category: 'bedroom',
  delivery_type: 'K',
  list_price: 49.99,
  price: 49.99,
  quantity: 2,
  sku: '50222606',
  title: 'Twin Bunkie Boards',
}

export const mockActiveAddon = {
  age: ['child'],
  breadcrumb_url: '/furniture/kids-and-teens',
  catalog: 'kids',
  catalog_availability: [{ FL: true }, { OOM: true }, { SE: true }, { TX: true }],
  category: 'bedroom',
  createdAt: '2019-07-03T16:44:10.171Z',
  delivery_type: 'K',
  description: 'Twin Bunkie Boards Twin Bunkie Boards',
  free_shipping: false,
  generic_name: 'twin bunkie boards',
  grid_image:
    'https://images.rtg-dev.com/twin-bunkie-boards_50222606_grid-image-item?cache-id=1a3495eb7cd46f299bce08a82dffd38f',
  high_res_image:
    'https://images.rtg-dev.com/twin-bunkie-boards_50222606_high-res-image-item?cache-id=ce8aac5dc0808e4139d59d4ef5d2b95e',
  lastModified: '2019-07-03T16:44:10.171Z',
  monogram: false,
  mpn: '502-CS106',
  on_sale: [
    { FL_0: false },
    { FL_3: false },
    { FL_4: false },
    { FL_6: false },
    { SE_0: false },
    { SE_2: false },
    { SE_3: false },
    { SE_4: false },
    { TX_0: false },
    { TX_4: false },
    { TX_5: false },
    { TX_6: false },
  ],
  pricing: [
    { FL_0_list_price: 49.99 },
    { FL_0_sale_price: 49.99 },
    { FL_3_list_price: 49.99 },
    { FL_3_sale_price: 49.99 },
    { FL_4_list_price: 49.99 },
    { FL_4_sale_price: 49.99 },
    { FL_6_list_price: 49.99 },
    { FL_6_sale_price: 49.99 },
    { SE_0_list_price: 49.99 },
    { SE_0_sale_price: 49.99 },
    { SE_2_list_price: 49.99 },
    { SE_2_sale_price: 49.99 },
    { SE_3_list_price: 49.99 },
    { SE_3_sale_price: 49.99 },
    { SE_4_list_price: 49.99 },
    { SE_4_sale_price: 49.99 },
    { TX_0_list_price: 49.99 },
    { TX_0_sale_price: 49.99 },
    { TX_4_list_price: 49.99 },
    { TX_4_sale_price: 49.99 },
    { TX_5_list_price: 49.99 },
    { TX_5_sale_price: 49.99 },
    { TX_6_list_price: 49.99 },
    { TX_6_sale_price: 49.99 },
    { default_price: 49.99 },
  ],
  primary_image:
    'https://images.rtg-dev.com/twin-bunkie-boards_50222606_image-item?cache-id=fbd26fd0a99f3b1e470cd31d00143223',
  quantity: 2,
  sell_individually: true,
  shipping_cost_code: [{ FL: 'ITEM' }, { SE: 'ITEM' }, { TX: 'ITEM' }],
  single_item_room: false,
  size: 'twin',
  sku: '50222606',
  sub_category: ['Bunkie Boards'],
  swatch_image:
    'https://images.rtg-dev.com/twin-bunkie-boards_50222606_swatch-image-item?cache-id=f69172c21fe719d82fbb41982a2213f7',
  swatch_request_flag: false,
  title: 'Twin Bunkie Boards',
  type: 'item',
  variation_keys: [],
}
