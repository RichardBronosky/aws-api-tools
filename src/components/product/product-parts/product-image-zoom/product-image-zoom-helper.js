import { announce, taskDone } from '@helpers/aria-announce'

export const handleWindowEvents = e => {
  if (window.zoom) {
    if (e.key == 'Escape' || e.key == 'Enter' || e.keyCode == 32) {
      e.preventDefault()
      window.setZoom(false)
      if (window.zoomBtn && window.zoomBtn.current) {
        window.zoomBtn.current.setAttribute('aria-pressed', 'false')
        taskDone(
          () => {
            window.zoomBtn.current.focus()
          },
          500,
          'focusZoom'
        )
      }
    }
  }
}

export const handleZoomToggle = (e, zoom, setZoom, zoomBtn) => {
  if (!e) return
  e.stopPropagation()
  if (e.type == 'click' || e.key == 'Enter' || e.keyCode == 32 || !e.keyCode) {
    e.preventDefault()
    setZoom(zoom)
    if (zoomBtn && zoomBtn.current) {
      zoomBtn.current.setAttribute('aria-pressed', zoom)
    }
    if (zoom) {
      announce('Image Zoomed')
      window.addEventListener('keydown', handleWindowEvents)
      window['zoom'] = zoom
      window['setZoom'] = setZoom
      window['zoomBtn'] = zoomBtn
    } else {
      announce('Image Zoom Closed')
      window.removeEventListener('keydown', handleWindowEvents)
    }
  }
  if (e.key == 'Escape') {
    window.removeEventListener('keydown', handleWindowEvents)
    announce('Image Zoom Closed')
    setZoom(false)
    if (zoomBtn && zoomBtn.current) {
      zoomBtn.current.setAttribute('aria-pressed', 'false')
      taskDone(
        () => {
          zoomBtn.current.focus()
        },
        500,
        'focusZoom'
      )
    }
  }
}
