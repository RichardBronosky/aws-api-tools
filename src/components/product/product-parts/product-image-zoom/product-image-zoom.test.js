import React from 'react'
import renderer from 'react-test-renderer'
import { mockProduct1 } from '@mocks/productDataMocks'
import ProductImageZoomWrapper from './index'
import ProductImageZoom from './product-image-zoom'
import * as imageZoomHelper from './product-image-zoom-helper'
import * as announceHelper from '@helpers/aria-announce'

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

describe('ProductImageZoom', () => {
  const testProps = {
    image: 'www.test-imagesrc.rtg.com/image.png',
    product: mockProduct1,
    zoom: false,
    setZoom: jest.fn(),
    zoomBtnRef: '',
    isMobile: true,
  }

  it('renders correctly with product and image', () => {
    const tree = renderer.create(<ProductImageZoomWrapper { ...testProps } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with product and image when zoomed', () => {
    const tree = renderer.create(<ProductImageZoom { ...{ ...testProps, zoom: true } } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('handleZoomToggle calls setZoom and annouce when enter pressed', () => {
    announceHelper.taskDone = jest.fn()
    announceHelper.announce = jest.fn()
    imageZoomHelper.handleZoomToggle(
      { stopPropagation: jest.fn(), preventDefault: jest.fn(), key: 'Enter' },
      true,
      testProps.setZoom,
      {
        current: {
          setAttribute: jest.fn(),
          focus: jest.fn(),
        },
      }
    )
    expect(testProps.setZoom).toHaveBeenCalled()
    expect(announceHelper.announce).toHaveBeenCalled()
  })

  it('handleZoomToggle calls setZoom, taskDone, and annouce when escape pressed', () => {
    announceHelper.taskDone = jest.fn()
    announceHelper.announce = jest.fn()
    imageZoomHelper.handleZoomToggle(
      { stopPropagation: jest.fn(), preventDefault: jest.fn(), key: 'Escape' },
      true,
      testProps.setZoom,
      {
        current: {
          setAttribute: jest.fn(),
          focus: jest.fn(),
        },
      }
    )
    expect(testProps.setZoom).toHaveBeenCalled()
    expect(announceHelper.announce).toHaveBeenCalled()
    expect(announceHelper.taskDone).toHaveBeenCalled()
  })

  it('calls handZoomToggle when enter pressed', () => {
    imageZoomHelper.handleZoomToggle = jest.fn()
    const tree = renderer.create(<ProductImageZoom { ...testProps } />)
    tree.root.findByProps({ className: 'product-image cell small-10' }).props.onKeyDown({ key: 'Enter' })
    expect(imageZoomHelper.handleZoomToggle).toHaveBeenCalled()
  })

  it('calls handZoomToggle when clicked', () => {
    imageZoomHelper.handleZoomToggle = jest.fn()
    const tree = renderer.create(<ProductImageZoom { ...testProps } />)
    tree.root.findByProps({ className: 'product-image cell small-10' }).props.onClick()
    expect(imageZoomHelper.handleZoomToggle).toHaveBeenCalled()
  })

  it('handleWindowEvents calls taskDone', () => {
    announceHelper.taskDone = jest.fn()
    global['zoom'] = true
    global['setZoom'] = jest.fn()
    global['zoomBtn'] = {
      current: {
        setAttribute: jest.fn(),
      },
    }
    imageZoomHelper.handleWindowEvents({ preventDefault: jest.fn(), key: 'Enter' })
    expect(announceHelper.taskDone).toHaveBeenCalled()
  })
})
