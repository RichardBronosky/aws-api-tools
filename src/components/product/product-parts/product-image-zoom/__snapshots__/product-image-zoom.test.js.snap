// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImageZoom renders correctly with product and image 1`] = `
<div
  aria-describedby="imageZoomInstructions"
  aria-pressed="false"
  className="product-image cell small-10"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <img
    alt="Adaire Brown Side Table"
    className="small-image"
    src="www.test-imagesrc.rtg.com/image.png&h=550"
  />
  <img
    alt="zoom icon"
    className="zoom-icon"
    src={Object {}}
  />
</div>
`;

exports[`ProductImageZoom renders correctly with product and image when zoomed 1`] = `
Array [
  <div
    aria-describedby="imageZoomInstructions"
    aria-pressed="false"
    className="product-image cell small-10"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    <img
      alt="Adaire Brown Side Table"
      className="small-image"
      src="www.test-imagesrc.rtg.com/image.png&h=550"
    />
    <img
      alt="zoom icon"
      className="zoom-icon"
      src={Object {}}
    />
  </div>,
  <div />,
]
`;
